x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -933,28 +942,115 @@ export class DataRecognizer {
     }
   }
 
-  // ensure the model memory limit for each job is not greater than
-  // the max model memory setting for the cluster
-  async updateModelMemoryLimits(moduleConfig: Module) {
+  /**
+   * Provides a time range of the last 3 months of data
+   */
+  async getFallbackTimeRange(
+    timeField: string,
+    query?: any
+  ): Promise<{ start: number; end: number }> {
+    const fieldsService = fieldsServiceProvider(this.callAsCurrentUser);
+
+    const timeFieldRange = await fieldsService.getTimeFieldRange(
+      this.indexPatternName,
+      timeField,
+      query
+    );
+
+    return {
+      start: timeFieldRange.end.epoch - moment.duration(3, 'months').asMilliseconds(),
+      end: timeFieldRange.end.epoch,
+    };
+  }
+
+  /**
+   * Ensure the model memory limit for each job is not greater than
+   * the max model memory setting for the cluster
+   */
+  async updateModelMemoryLimits(
+    moduleConfig: Module,
+    estimateMML: boolean = false,
+    start?: number,
+    end?: number
+  ) {
     const { limits } = await this.callAsCurrentUser('ml.info');
+
     const maxMml = limits.max_model_memory_limit;
-    if (maxMml !== undefined) {
-      // @ts-ignore
-      const maxBytes: number = numeral(maxMml.toUpperCase()).value();
-
-      if (Array.isArray(moduleConfig.jobs)) {
-        moduleConfig.jobs.forEach(job => {
-          const mml = job.config?.analysis_limits?.model_memory_limit;
-          if (mml !== undefined) {
-            // @ts-ignore
-            const mmlBytes: number = numeral(mml.toUpperCase()).value();
-            if (mmlBytes > maxBytes) {
-              // if the job's mml is over the max,
-              // so set the jobs mml to be the max
-              job.config.analysis_limits!.model_memory_limit = maxMml;
-            }
+    // @ts-ignore
+    const maxBytes: number = numeral(maxMml.toUpperCase()).value();
+
+    if (!Array.isArray(moduleConfig.jobs)) {
+      return;
+    }
+
+    if (estimateMML && this.jobsForModelMemoryEstimation.length > 0) {
+      const calculateModelMemoryLimit = calculateModelMemoryLimitProvider(this.callAsCurrentUser);
+      const query = moduleConfig.query ?? null;
+
+      const jobs = moduleConfig.jobs.filter(job =>
+        this.jobsForModelMemoryEstimation.includes(job.id)
+      );
+
+      // Checks if all jobs in the module have the same time field configured
+      const isSameTimeFields = jobs.every(
+        job => job.config.data_description.time_field === jobs[0].config.data_description.time_field
+      );
+
+      if (isSameTimeFields && (start === undefined || end === undefined)) {
+        // In case of time range is not provided and the time field is the same
+        // set the fallback range for all jobs
+        const { start: fallbackStart, end: fallbackEnd } = await this.getFallbackTimeRange(
+          jobs[0].config.data_description.time_field,
```

Comment:
this causes a crash, `jobs` could be empty at this point.

Reply:
hm, how come? 🤔 there is a check for `this.jobsForModelMemoryEstimation.length > 0` before

Reply:
It looks like the job list creation isn't using the job prefix when comparing ids